### PR TITLE
Fix passing Isolate between c and rust 

### PR DESCRIPTION
**Pull request by piscisaureus**
I'm not super sure about the rename from `data` to `isolate_data`, it is still very ambiguous. 
Maybe it should be `callback_data` or `user_data` or `embedder_data` instead.

### CODE REVIEW

File: libdeno/binding.cc [hunk 1]
```diff
@@ -401,6 +401,24 @@ void AddIsolate(Deno* d, v8::Isolate* isolate) {
   d->isolate->SetData(0, d);
 }
 
+class IsolateDataScope {
+  Deno* deno;
+  void* prev_data;
+  void* data;  // Not necessary; only for sanity checking.
```
Comment (ry) [inline libdeno/binding.cc]: 👍

File: Roadmap.md [hunk 2]
```diff
@@ -131,7 +131,7 @@ void deno_set_callback(Deno* deno, deno_sub_cb cb);
 // Get error text with deno_last_exception().
 // 0 = success, non-zero = failure.
 // TODO(ry) Currently the return code has opposite semantics.
-int deno_execute(Deno* d, const char* js_filename, const char* js_source);
+int deno_execute(Deno* d, void* isolate_data, const char* js_filename, const char* js_source);
```
[the remark this answers is not in the document]
Comment (piscisaureus) [reply Roadmap.md]: I think it looks very conflicting here:

```c
typedef void (*deno_recv_cb)(void* isolate_data, int32_t req_id,
                             deno_buf control_buf, deno_buf data_buf);
```

**Comment by piscisaureus** [comment]
I'm gonna change isolate_c again to libdeno_isolate.

File: Roadmap.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (piscisaureus) [reply Roadmap.md]: I think I like user_data better too.

**Comment by piscisaureus** [comment]
> I’m surprised there’s no change to main.rs. If I understand this correctly the rust isolate is stored on the stack in main() ?

It used to be in a Box, so the Isolate itself would be heap-allocated.
Now it likely does end up on the stack, although that's for really for the rust compiler to decide -- it could start out on the stack and move to the heap later.

**Pull request merged**